src/core/frontend/image_interface.h

```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 Citra Emulator Project
+// Licensed under GPLv2 or any later version
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include "common/common_types.h"
+#include "common/logging/log.h"
+
+namespace Frontend {
+
+class ImageInterface {
+public:
+    ImageInterface() = default;
+    ~ImageInterface() = default;
+
+    // Error logging should be handled by the frontend
+    virtual bool DecodePNG(std::vector<u8>& dst, u32& width, u32& height, const std::string& path) {
```

Comment:
```suggestion
    virtual bool DecodePNG(std::vector<u8>& dst, u32& width, u32& height, const std::string& path) = 0;
```
make it pure virtual